### PR TITLE
Validator

### PR DESCRIPTION
**Pull request by stephenmuller**

### CODE REVIEW

**Comment by champs794** [comment]
Took me a second to realize this was the pull request for Validator.

File: practice/validator/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <meta charset="utf-8">
+    <title>Validator.</title>
+  </head>
+  <body>
+    <form>
+        Full Name (First, Last):
+      <input id="full-name" type="text" name="fullName" placeholder="First, Last">
+        Date of Birth (1900-12-30):
+      <input id="birth-date" type="text" name="birthdate" placeholder="year-month-day">
+      Phone number (123-456-7890):
```
Comment (champs794) [inline practice/validator/index.html]: Remember to use the <label> tag, which makes input fields bigger/easier targets for finger taps and mouse clicks.

It's both a convenience and an accommodation for people with various disabilities that prevent them from doing it the "normal" way.

File: practice/validator/site.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+
+/**
+*event handlers
+*/
+function registerEventHandlers() {
+  $('#full-name').on('keyup', nameHandler);
+  $('#birth-date').on('keyup', birthdateHandler);
+  $('#phone-number').on('keyup', phoneNumberHandler);
```
Comment (champs794) [inline practice/validator/site.js]: Corey noted that there is an even better event: *input*

File: practice/validator/site.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  testValueForBackgroundColor(toTest, regexpTest);
+}
+
+/**
+* handles phone# field validation
+*/
+function phoneNumberHandler() {
+  var regexpTest = /^\d{3}\-\d{3}\-\d{4}$/;
+  var toTest = $('#phone-number');
+  testValueForBackgroundColor(toTest, regexpTest);
```
Comment (champs794) [inline practice/validator/site.js]: Try breaking these handlers into smaller steps. As written, there is no way to test validation with an arbitrary value;

Comment (selassid) [inline practice/validator/site.js]: I agree with this. Yes, a `testValueForBackgroundColor()` function that takes the element and the regexp results in the most concise code, it's harder to test the core operation: is this string valid?

File: practice/validator/site.js [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
-  if (testKey.test(fieldPointer.val())) {
-    $(fieldPointer).removeClass('invalid').addClass('valid');
+function setFieldBackground(fieldPointer, testKey) {
+  if (testValue(fieldPointer, testKey)) {
+    setValid(fieldPointer);
   } else {
-    $(fieldPointer).removeClass('valid').addClass('invalid');
+    setInvalid(fieldPointer);
   }
 }
```
Comment (selassid) [inline practice/validator/site.js]: These are some good improvements, as some of the steps are now in named functions, but I don't think it addresses the previous concern: these functions are still difficult to theoretically doctest, and mix "input", "transform", and "main" steps all up. There's still no function I can call to test that an arbitrary phone number is valid.

I think this code might be valuing conciseness over testability and clarity. I think you should spend some distance from this problem and come back to thinking about its structure once you've worked on some other stuff for a while.

**Pull request merged**